File: scrape.py
import re

from urllib.request import urlopen
from bs4 import BeautifulSoup

query = "bread"
url = "https://www.google.com/finance?noIL=1&q=" + query
web_page = urlopen(url)
soup = BeautifulSoup(web_page, 'html.parser')

mytds = soup.findAll("td", { "class" : "symbol" })
names = []
for td in mytds:
    ## 3 or 4 letter symbols
    result = re.search(r'[A-Z][A-Z][A-Z][A-Z]?', td.text)
    if(result != None):
        names.append(result.group())
print(names)

File: scrape.js
const request = require('request');
var cheerio = require('cheerio');

function Scrape(){
  var names = [];
  var query = "bread"; // Hardcode for now
  var url = "https://www.google.com/finance?noIL=1&q=" + query;

  request(url, function (error, response, html) {
    if (!error && response.statusCode == 200) {
        var $ = cheerio.load(html);
        $('td.symbol').each(function(i, element){
          var a = $(this);
          var ticker = /[A-Z][A-Z][A-Z][A-Z]?/
        var res = ticker.exec(a.text());
          if(res != null){
            names.push(res[0]);
            console.log(res[0]);
            }
        });
      }
    console.log(names);
  });
  return names;
}


